test/compile_test.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+# Run this test from the root of the repository, as:
+# $ bazel test :all --test_output=errors
+
+import unittest
+import subprocess
+import difflib
+import os
+import re
+
+
+class TestSourcesMatchCommittedCSV(unittest.TestCase):
+  """Check that what's committed matches the output of the compiler.
+
+  Any change to this repository that modifies sources for some country's data
+  (the stuff in identifiers/*/*.csv) should also modify the corresponding
+  compiled output (identifiers/country-??.csv), such that they match.
+
+  This test runs the compiler, and compares its output to what's been
+  committed, ensuring they are equal. If this test fails, you need to re-run
+  the compiler for your country, and commit the result.
+  """
+
+  def get_committed_and_compiled_data(self, country_code):
+    # Read what's in the repo.
+    committed_csv_path = F'identifiers/country-{country_code}.csv'
+    with open(committed_csv_path, 'r') as committed_file:
+      committed_csv = committed_file.read()
+
+    # Run the compiler with temp output.
+    compiler_output_path = os.path.join(os.getenv('TEST_TMPDIR'), F"out-{country_code}.csv")
+    subprocess.run([
```

Comment:
I haven't tried this out, but what happens if the directory of country specific files hasn't been created. Does the compiler fail and the test fail to run? Or do we still get a sane output?

I think ideally we'd be able to just tell the user that there's a mismatch

Reply:
Currently, if `identifiers/country-xx.csv` exists, but `identifiers/country-xx/` does not:

- The test fails, asking you to run `./scripts/compile.py xx`
- Running that command will overwrite `identifiers/country-xx.csv` with a blank file, which is probably not what we want.

I'll see if I can update the test to ignore country xx in this case.

Reply:
Done.